### PR TITLE
Multiple fixes

### PR DESCRIPTION
**Pull request by shivasishdas**
- Adds strict timeout for the push on endpoint as long timeouts were causing consumer group to miss heartbeat and get evicted from the broker
`
{"level":"error","ts":1618949976.4781258,"caller":"worker/stream.go:133","msg":"worker: error posting messages to subscription url","gitCommitHash":"00cd00e5276c517ab34971c16694ec527acf1c50","appEnv":"","serviceName":"metro","error":"Post \"https://webhook.site/e648d6c6-9f08-488a-bdc9-6c956169e991\": dial tcp: i/o timeout","stacktrace":"github.com/razorpay/metro/service/worker.(*PushStream).processPushStreamResponse\n\t/src/service/worker/stream.go:133\ngithub.com/razorpay/metro/service/worker.(*PushStream).Start.func3\n\t/src/service/worker/stream.go:94\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57"}
`

`%4|1618949972.383|SESSTMOUT|rdkafka#consumer-12| [thrd:main]: Consumer group session timed out (in join-state started) after 24192 ms without a successful response from the group coordinator (broker 2, last error was Success): revoking assignment and rejoining group
%5|1618949975.346|REQTMOUT|rdkafka#consumer-11| [thrd:GroupCoordinator]: GroupCoordinator/2: Timed out HeartbeatRequest in flight (after 1985ms, timeout #0)
%5|1618949975.346|REQTMOUT|rdkafka#consumer-8| [thrd:GroupCoordinator]: GroupCoordinator/1: Timed out HeartbeatRequest in flight (after 1616ms, timeout #0)`

- Change Kafka produce message timeout to 500ms from 50ms.  At high TPS(50 VUS) produce API hits, produce calls to the broker was failing with frequent timeouts.
`failed to produce message to topic [projects/perfproject/topics/perftopic1] due to timeout [50]`

- Init common HTTP client on push stream init
- More metrics around errors and push stream
- Change `auto.offset.reset kakfa` consumer to latest.
- Add retry on Commit operation timeouts
- More logs

### CODE REVIEW

File: internal/publisher/core.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func NewCore(bs brokerstore.IBrokerStore) *Core {
 
 // Publish messages
 func (p *Core) Publish(ctx context.Context, req *metrov1.PublishRequest) ([]string, error) {
-	producer, err := p.bs.GetProducer(ctx, messagebroker.ProducerClientOptions{Topic: req.Topic, TimeoutMs: 50})
+	producer, err := p.bs.GetProducer(ctx, messagebroker.ProducerClientOptions{Topic: req.Topic, TimeoutMs: 500})
```
Comment (Abhishekvrshny) [inline internal/publisher/core.go:26]: lets not do this. We'll figure out the bottlenecks.

File: service/worker/stream.go [hunk 2]
```diff
@@ -121,14 +122,21 @@ func (ps *PushStream) Stop() error {
...
 	for _, message := range data.ReceivedMessages {
 		logger.Ctx(ps.ctx).Infow("worker: publishing response data to subscription endpoint")
 		if message.AckId == "" {
 			continue
 		}
 
 		postData := bytes.NewBuffer(message.Message.Data)
-		resp, err := http.Post(subModel.PushEndpoint, "application/json", postData)
+		// TODO: move http client creation during push stream init
+		httpClient := &http.Client{
```
Comment (Abhishekvrshny) [inline service/worker/stream.go:134]: httpclient can be cached. Also, make sure the transport options are optimised.

**Comment by Abhishekvrshny** [comment]
undo the timeout changes and fix build failure

File: service/worker/stream.go (hunk 2 above)
Comment (shivasishdas) [reply service/worker/stream.go]: Yup. Had left a TODO for that. Changed the impl. Moved HTTP init inside new push stream and re-using that client.

**Review by sunnyagg — APPROVED** [review]

**Review by Abhishekvrshny — APPROVED** [review]

**Review by sunnyagg — APPROVED** [review]

**Pull request merged**